lib/board.rb

```diff
@@ -62,11 +79,11 @@ def win_with_diagonal?(mark)
   end
 
   def check_forward_diagonal?(mark)
-    [0, 1, 2].all? { |index| board[index][index] == mark }
+    board.size.times.all? { |index| board[index][index] == mark }
   end
 
   def check_backward_diagonal?(mark)
-    [0, 1, 2].all? { |index| board[index][backward_index(index)] == mark }
+    board.size.times.all? { |index| board[index][backward_index(index)] == mark }
   end
 
   def backward_index(index)
```

Comment:
I think it would be worth adding a couple of tests to make sure a winning row is found on a 4x4 board.